main/src/main/java/org/mobilitydata/gtfsvalidator/report/model/FeedMetadata.java

```diff
@@ -0,0 +1,131 @@
+package org.mobilitydata.gtfsvalidator.report.model;
+
+import java.util.*;
+import java.util.function.Function;
+import org.mobilitydata.gtfsvalidator.table.*;
+
+public class FeedMetadata {
+  private Map<String, TableMetadata> tableMetaData;
+  public Map<String, Integer> counts = new TreeMap<>();
+
+  public Map<String, String> feedInfo = new LinkedHashMap<>();
+
+  public Map<String, String> specFeatures = new LinkedHashMap<>();
+
+  public ArrayList<AgencyMetadata> agencies = new ArrayList<>();
+
+  public static FeedMetadata from(GtfsFeedContainer feedContainer) {
```

Comment:
suggestion: this method populates all the necessary fields and counts. I suggest making the class-default constructor protected to avoid instantiation and prevent unexpected behaviors while calling methods in the class without proper field initialization.